haas/ext/switches/brocade.py

```diff
@@ -277,8 +277,18 @@ def _construct_tag(name):
         """ Construct the xml tag by prepending the brocade tag prefix. """
         return '{urn:brocade.com:mgmt:brocade-interface}%s' % name
 
-    def _make_request(self, method, url, data=None):
+    def _make_request(self, method, url, data=None, log=True):
         r = requests.request(method, url, data=data, auth=self._auth)
-        if r.status_code >= 400:
+
+        # error logging
+        # remove/delete something that's not attached/exists
+        if ((method == 'DELETE') and (not log) and (r.status_code == 404)):
```

Comment:
Because the caller method understands the request it is making, ``_make_request`` should just be a wrapper around making a generic request. And shouldn't concern itself with that logic.

```python
r = self._make_request('DELETE', url, log=False)
if r.status_code not in [200, 404]:
    logger.error('Bad Request to switch. Response: %s' % r.text)
```

Alternatively, and probably a better approach is to change ``_make_request`` to accept a list of acceptable error codes:

```python
def _make_request(self, method, url, data=None, acceptable_error_codes=None):
    acceptable_errors = acceptable_errors or []
    r = requests.request(method, url, data=data, auth=self._auth)
    if r.status_code >= 400 and r.status_code not in acceptable_error_codes:
        logger.error('Bad Request to switch. Response: %s' % r.text)
    return r
```

and call it from as ``self._make_request('DELETE', url, acceptable_error_codes=[404])``